treeherder/webapp/api/performance_serializers.py

```diff
@@ -398,26 +398,17 @@ def get_name(self, value):
 
 
 class PerfCompareResultsQueryParamsSerializer(serializers.Serializer):
-    startday = serializers.DateTimeField(required=False, allow_null=True, default=None)
-    endday = serializers.DateTimeField(required=False, allow_null=True, default=None)
     base_revision = serializers.CharField(required=False, allow_null=True, default=None)
-    new_revision = serializers.CharField(required=False, allow_null=True, default=None)
+    new_revision = serializers.CharField()
     base_repository = serializers.CharField()
     new_repository = serializers.CharField()
     framework = serializers.IntegerField(required=False, allow_null=True, default=None)
     interval = serializers.IntegerField(required=False, allow_null=True, default=None)
     no_subtests = serializers.BooleanField(required=False)
 
     def validate(self, data):
-        if (
-            data['base_revision'] is None
-            and data['new_revision'] is None
-            and data['interval'] is None
-            and (data['startday'] is None or data['endday'] is None)
-        ):
-            raise serializers.ValidationError(
-                'Required: base_revision, new_revision, startday and endday or interval.'
-            )
+        if data['base_revision'] is None and data['interval'] is None:
```

Comment:
`interval` field is not required when the base revision is provided, but if it's not provided then it needs to be sent in the URL param so that we can calculate further the start_day and end_day, which will be used to filter the data for the base repository.